doc/sphinxext/gallery_generator.py

```diff
@@ -282,20 +323,41 @@ def extract_docstring(self):
             break
 
         thumbloc = None
-        for i, line in enumerate(docstring.split("\n")):
+        title: Optional[str] = None
+        ex_title: str = ""
+        for line in docstring.split("\n"):
+            # capture the first non-empty line of the docstring as title
+            if ex_title == "":
+                ex_title = line
```

Comment:
do you want a `continue` here, to avoid setting the title to `_thumb: 0.3, 0.6` or whatever?

Reply:
Actually, we can still end up setting the title to `_thumb`, even with that change.  To do what you suggest, I have to reorder the clauses in the body of the for loop. I'll do that now.